### ISSUE TITLE
Release 0.9

### ISSUE THREAD

**Issue by Huxpro** [facebook/hermes#586]
# Hermes 0.9.0 for React Native 0.66

Related RN links:
- https://github.com/react-native-community/releases/issues/246

### Highlighted Changes

This release is primarily made to close the gap between the Hermes cut and the (upcoming) React Native 0.66 cut. Among the ~400 commits, we kept delivering memory and size wins, bugfixes, and made other progress behind the scene.

### Checklist

- [x] Regression tests (on CI-built artifacts)
  - [x] Android 
  - [x] iOS
- [x] Publish on Github
  - [x] Highlights (Release Notes)
- [x] Publish to NPM
- [x] Publish to CocoaPod

### File Pick Requests in RN repo

- [x] the `libjsi` fix
- [x] the GCConfig fix
- [x] Hermes version bump

### Regression tests and releases logs

<details>
<summary>Android</summary>

![image](https://user-images.githubusercontent.com/5563315/131757174-8a9e46a6-f9ed-495c-b63e-b83c7480a684.png)

</details>

<details>
<summary>iOS</summary>

![image](https://user-images.githubusercontent.com/5563315/131801678-922db341-6eea-479d-bc8d-4c33c697de5a.png)

</details>

<details>
<summary>Release</summary>

- NPM

```
npm notice === Tarball Details ===
npm notice name:          hermes-engine
npm notice version:       0.9.0
npm notice filename:      hermes-engine-0.9.0.tgz
npm notice package size:  15.4 MB
npm notice unpacked size: 20.6 MB
npm notice shasum:        84d9cfe84e8f6b1b2020d6e71b350cec84ed982f
npm notice integrity:     sha512-r7U+Y4P2Qg/ig[...]U19RKgWbr3qPw==
npm notice total files:   48
npm notice
+ hermes-engine@0.9.0
```

```
npm notice === Tarball Details ===
npm notice name:          hermes-engine-cli
npm notice version:       0.9.0
npm notice filename:      hermes-engine-cli-0.9.0.tgz
npm notice package size:  52.0 MB
npm notice unpacked size: 126.0 MB
npm notice shasum:        f6d8d7037136f16d8c2e467a80b83dd640cd40a9
npm notice integrity:     sha512-un0lmLPuS65u1[...]ApJFl/6FYMHlA==
npm notice total files:   27
npm notice
+ hermes-engine-cli@0.9.0
```

- Cocoapod

```
--------------------------------------------------------------------------------
 🎉  Congrats

 🚀  hermes-engine (0.9.0) successfully published
 📅  September 2nd, 00:39
 🌎  https://cocoapods.org/pods/hermes-engine
 👍  Tell your friends!
--------------------------------------------------------------------------------
```

</details>

**Comment by Huxpro** [comment]
Made a pick request on https://github.com/facebook/react-native/commit/1bc885b for RN 0.66-rc2. Otherwise `libjsi` would be failed to link on Android.

**Comment by chishing96** [comment]
there's vulnerability issue from hermes 0.9.0 CVE-2020-1914 which was fixed [here](https://github.com/facebook/hermes/commit/379760bbc5a27802917ac9e7847da356df9c9660) 

how do we update this fix for our RN 0.66.1 project

**Comment by neildhar** [comment]
@chishing96 This commit is already in 0.9 as https://github.com/facebook/hermes/commit/b2021df620824627f5a8c96615edbd1eb7fdddfc.
